### PR TITLE
use nsenter to access container namespace

### PR DESCRIPTION
**Pull request by phoracek**
With Go 1.9 or older, namespaces does not work properly. With this patch we execute a separate process (nsenter command line) every time we access container namespace.

### CODE REVIEW

**Comment by phoracek** [comment]
@yuvalif @SchSeba hi, could you please take a look?

File: pkg/plugin/plugin.go [hunk 1]
```diff
@@ -266,13 +266,10 @@ func CmdDel(args *skel.CmdArgs) error {
 
 	// Delete can be called multiple times, so don't return an error if the
 	// device is already removed.
-	err = ns.WithNetNSPath(args.Netns, func(ns.NetNS) error {
-		err = ip.DelLinkByName(args.IfName)
-		if err != nil && err == ip.ErrLinkNotFound {
-			return nil
-		}
+	_, err = withNetNS(args.Netns, "ip", "link", "del", "dev", args.IfName)
+	if err != nil && strings.Contains(err.Error(), "Cannot find device") {
```
Comment (yuvalif) [inline pkg/plugin/plugin.go]: the logic is inverse, if you have an error which is not "cannot find device" then return an error

File: pkg/plugin/plugin.go [hunk 2]
```diff
@@ -177,11 +182,12 @@ func CmdAdd(args *skel.CmdArgs) error {
 		return err
 	}
 
-	// Refetch the bridge since its MAC address may change when the first
-	// veth is added.
-	if err = refetchIface(brIface); err != nil {
-		return err
+	// Refetch the bridge MAC since it may change when the first veth is added.
```
Comment (yuvalif) [inline pkg/plugin/plugin.go]: how is this change related to the ns fix?

**Comment by yuvalif** [comment]
not sure regarding issue #6 
which supported platforms don't support golang10 ?

File: pkg/plugin/plugin.go [hunk 3]
```diff
@@ -86,32 +82,56 @@ func setupBridge(brName string) (*current.Interface, error) {
...
 
-	// Refetch the hostIface since its MAC address may change during network namespace move.
-	if err = refetchIface(hostIface); err != nil {
+	contIface.Name = contIfaceName
+	contIface.Sandbox = contNetnsPath
+	contIfaceMac, err := getIfaceMac(contNetnsPath, contIfaceName)
+	if err != nil {
 		return nil, nil, err
 	}
+	contIface.Mac = contIfaceMac
```
Comment (SchSeba) [inline pkg/plugin/plugin.go]: small nit don't need a new variable here just use `contIface.Mac`

File: pkg/plugin/plugin.go (hunk 1 above)
Comment (SchSeba) [reply pkg/plugin/plugin.go]: Need to be `if err != nil && !(strings.Contains(err.Error(), "Cannot find device"))`

**Comment by phoracek** [comment]
This repository master will require Go 1.10, 1.9 will be handled on a separate private branch to not block further development.
